.github/workflows/nix.yml

```diff
@@ -47,19 +69,40 @@ jobs:
           conan profile update env.CC=${{ matrix.profile.cc }} default
           conan profile update env.CXX=${{ matrix.profile.cxx }} default
           conan profile update conf.tools.build:compiler_executables='{"c": "${{ matrix.profile.cc }}", "cpp": "${{ matrix.profile.cxx }}"}' default
-      - name: checkout
-        uses: actions/checkout@v3
-      - name: dependencies
+          # Do not quote the URL. An empty string will be accepted (with
+          # a non-fatal warning), but a missing argument will not.
+          conan remote add ripple ${{ env.CONAN_URL }} --insert 0
+      - name: try to authenticate to ripple Conan remote
+        id: remote
+        continue-on-error: true
+        run: |
+          conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }}
```

Comment:
As discussed elsewhere, to avoid the confusing `Process completed with exit code 1` annotation on the summary screen, it may be worth rewriting this to set a status variable and so that this step always succeeds, then check that that variable in the later steps.

e.g.
```
echo outcome=$(conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }} && echo "success" || echo "failure") | tee ${GITHUB_OUTPUT}
```
Then the checks would be `steps.remote.outputs.outcome == 'success')`, etc.